pandas/core/series.py

```diff
@@ -3208,7 +3208,11 @@ def _try_cast(arr, take_fast_path):
         return subarr
 
     elif isinstance(data, (list, tuple)) and len(data) > 0:
-        if dtype is not None:
+        if all(is_list_like(item) for item in data):
```

Comment:
not sure this is the appropriate place for this.

Reply:
(let me try to persuade you that _other things_ were out of place!)

Reply:
For performance, can't we first try the `_try_cast` and only if that fails / does not give an appropriate sized subarr do this check of all lists?

Reply:
Not sure this would be simple to do... but anyway, notice that
- if ``data`` is "sane" (e.g. first element is not a list), then that line has a very small performance cost, since it is lazily evaluated
- while if ``data`` _is_ a list of lists, then going through ``_try_cast`` could have a performance cost (but most importantly from my point of view, behavior would be more complex to analyze)

Reply:
Just tested:

```
In [79]: %timeit np.array(data)
6.17 ms ± 53.3 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [80]: %timeit all(pd.api.types.is_list_like(item) for item in data)
2.2 µs ± 24.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

and the time for such checking is actually negligible compared to the time that the actual conversion to array takes. So indeed not needed for performance reasons.

(if it would not have been the case, I think keeping 'normal' list -> series fast would be higher in priority than the possible slowdown in converting list of lists)

Reply:
> (if it would not have been the case, I think keeping 'normal' list -> series fast would be higher in priority than the possible slowdown in converting list of lists)

I totally agree

Reply:
By the way: [if/when there will be](https://github.com/numpy/numpy/issues/5933#issuecomment-348930690) an ``ndmax`` argument to ``np.array``, it will allow us to speed up further the array creation.

Reply:
I would like to move this loggic to ``_init_listlike`` or something similar.